test/integration/com/unifina/service/PermissionServiceIntegrationSpec.groovy

```diff
@@ -308,13 +295,24 @@ class PermissionServiceIntegrationSpec extends IntegrationSpec {
 		service.systemGrantAll(me, canvas)
 
 		expect:
-		service.getPermissionsTo(stream, me).size() == 4
+		service.getPermissionsTo(stream, me).size() == 5
+		service.check(me, uiChannelStream, Permission.Operation.CANVAS_GET)
 		service.check(me, stream, Permission.Operation.STREAM_GET)
 		service.check(me, stream, Permission.Operation.STREAM_PUBLISH)
 		service.check(me, stream, Permission.Operation.STREAM_SUBSCRIBE)
 		service.check(me, stream, Permission.Operation.STREAM_DELETE)
 	}
 
+	void "getPermissionsTo(resource, userish) returns correct UI channel read permissions via associated dashboard"() {
+		service.systemGrantAll(me, dashboard)
+		def permissions = service.getPermissionsTo(uiChannelStream, me)
+
+		expect:
+		permissions.size() == 1
+		permissions.get(0).operation == Permission.Operation.CANVAS_GET
```

Comment:
How could the user have `CANVAS_GET` permission to a stream? Can't be right. It should return 2 permissions: `STREAM_GET` and `STREAM_SUBSCRIBE`.